azure-servicebus/src/main/java/com/microsoft/azure/servicebus/primitives/ConnectionStringBuilder.java

```diff
@@ -529,12 +593,34 @@ else if (key.equalsIgnoreCase(TRANSPORT_TYPE_CONFIG_NAME))
 							exception);
 				}
 			}
+			else if (key.equalsIgnoreCase(AUTHENTICATION_CONFIG_NAME))
+			{
+				this.authentication = AuthenticationType.ignoreCaseValueOf(values[valueIndex]);
+			}
 			else
 			{
 				throw new IllegalConnectionStringFormatException(
 						String.format(Locale.US, "Illegal connection string parameter name: %s", key));
 			}
-		}		
+		}
+		this.validate();
+	}
+
+	private void validate() {
+		// Without "aadAuthentication" field set to a valid value, other AAD fields should not be respected
+        boolean hasAuthentication = this.authentication != null;
+        boolean hasSasAuthentication = !StringUtil.isNullOrEmpty(this.sharedAccessKeyName);
+        boolean hasSharedAccessSignatureToken = !StringUtil.isNullOrEmpty(this.sharedAccessSingatureToken);
+
+        if (hasAuthentication && hasSasAuthentication)
+        {
+            throw new IllegalConnectionStringFormatException("Cannot have Authentication and SharedAccessKeyAuthentication simultaneously.");
```

Comment:
>Cannot have Authentication and SharedAccessKeyAuthentication simultaneously."); [](start = 62, length = 79)

Consider rephrasing this. Talk to a PM.

Reply:
How about "Cannot have values set for both X and Y" ?